lib/pure/future.nim

```diff
@@ -197,4 +197,32 @@ macro dump*(x: typed): untyped =
   let s = x.toStrLit
   let r = quote do:
     debugEcho `s`, " = ", `x`
-  return r
\ No newline at end of file
+  return r
+
+macro asArray*(targetType: typedesc, values: typed): untyped =
+  ## applies a type conversion to each of the elements in the specified
+  ## array literal. Each element is converted to the ``targetType`` type..
+  ##
+  ## Example:
+  ##
+  ## .. code-block::
+  ##   let x = asArray(int, [0.1, 1.2, 2.3, 3.4])
+  ##   doAssert x is array[4, int]
+  ##
+  ## Short notation for:
+  ##
+  ## .. code-block::
+  ##   let x = [(0.1).int, (1.2).int, (2.3).int, (3.4).int]
+  let tNode = getType(targetType)[1]
+  values.expectKind(nnkBracket)
+  result = newNimNode(nnkBracket, lineInfoFrom=values)
+  for i in 0 ..< len(values):
+    var dot = newNimNode(nnkDotExpr, lineInfoFrom=values[i])
```

Comment:
Do not produce a dot expression for the conversion but a call expression.

Reply:
done